apps/api/src/workspace-role/workspace-role.e2e.spec.ts

```diff
@@ -707,6 +861,26 @@ describe('Workspace Role Controller Tests', () => {
 
       expect(response.statusCode).toBe(401)
     })
+
+    it('should not be able to update the workspace role with environments that do not belong to the project', async () => {
+      const response = await app.inject({
+        method: 'PUT',
+        url: `/workspace-role/${adminRole1.slug}`,
+        payload: {
+          projectEnvironments: [
+            {
+              projectSlug: projects[0].slug,
+              environmentSlugs: ['production']
+            }
+          ]
+        },
+        headers: {
+          'x-e2e-user-email': alice.email
+        }
+      })
+
+      expect(response.statusCode).toBe(400)
+    })
   })
```

Comment:
Another case should be - should not be able to update workspace role where user doesn't have access to environment.